tasks/main.yml

```diff
@@ -7,8 +11,10 @@
   until: '"failed" not in result'
   retries: 5
   delay: 10
+  when: repofile is not defined
```

Comment:
`is not defined` would never be true, though—because no matter what the result of `stat` above, this check would always be `false`, since the variable would be defined.

Rather, can you change the clause (in both places) to:

```yaml
  when: epel_repofile.exists
```

(Also changing the name of the var to `epel_repofile`.)

Finally, could you make the default repo file path a variable defined inside `defaults/main.yml` instead of a hardcoded path?